go/vt/mysqlctl/binlogs_gtid.go

```diff
@@ -71,43 +71,42 @@ func ChooseBinlogsForIncrementalBackup(
 		if err != nil {
 			return nil, "", "", vterrors.Wrapf(err, "cannot get previous gtids for binlog %v", binlog)
 		}
-		prevPos, err := mysql.ParsePosition(mysql.Mysql56FlavorID, previousGtids)
+		previousGTIDsPos, err := mysql.ParsePosition(mysql.Mysql56FlavorID, previousGtids)
 		if err != nil {
-			return nil, "", "", vterrors.Wrapf(err, "cannot decode binlog %s position in incremental backup: %v", binlog, prevPos)
+			return nil, "", "", vterrors.Wrapf(err, "cannot decode binlog %s position in incremental backup: %v", binlog, previousGTIDsPos)
 		}
 		if prevGTIDsUnion == nil {
-			prevGTIDsUnion = prevPos.GTIDSet
+			prevGTIDsUnion = previousGTIDsPos.GTIDSet
 		} else {
-			prevGTIDsUnion = prevGTIDsUnion.Union(prevPos.GTIDSet)
+			prevGTIDsUnion = prevGTIDsUnion.Union(previousGTIDsPos.GTIDSet)
 		}
 
-		containedInFromPos := lookFromGTIDSet.Contains(prevPos.GTIDSet)
-		// The binary logs are read in-order. They are build one on top of the other: we know
-		// the PreviousGTIDs of once binary log fully cover the previous binary log's.
-		if containedInFromPos {
-			// All previous binary logs are fully contained by backupPos. Carry on
-			continue
-		}
-		// We look for the first binary log whose "PreviousGTIDs" isn't already fully covered
-		// by "backupPos" (the position from which we want to create the inreemental backup).
+		// The binary logs are read in-order. They expand. For example, we know
+		// Previous-GTIDs of binlog file 0000018 contain those of binlog file 0000017.
+		// We look for the first binary log whose Previous-GTIDs isn't already fully covered
+		// by "backupPos" (the position from which we want to create the incremental backup).
 		// That means the *previous* binary log is the first binary log to introduce GTID events on top
 		// of "backupPos"
-		if i == 0 {
-			return nil, "", "", vterrors.Errorf(vtrpc.Code_FAILED_PRECONDITION, "the very first binlog file %v has PreviousGTIDs %s that exceed given incremental backup pos. There are GTID entries that are missing and this backup cannot run", binlog, prevPos)
+		if backupFromGTIDSet.Contains(previousGTIDsPos.GTIDSet) {
+			// All previous binary logs are fully contained by backupPos. So definitely all binlogs _prior_ to
```

Comment:
IMO it's easier to reason about when explained as the current binary log being the oldest one available -- so all previous GTIDs have been purged and are no longer available.